staging/src/k8s.io/apiserver/pkg/cel/library/quantity_test.go

```diff
@@ -0,0 +1,276 @@
+			expr:        `quantity("200M") == quantity("200M")`,
+			expectValue: true,
+		},
+		{
+			name:        "equality_symmetry",
+			expr:        `quantity("200M") == quantity("0.2G") && quantity("0.2G") == quantity("200M")`,
+			expectValue: true,
+		},
+		{
+			name:        "equality_transitivity",
+			expr:        `quantity("2M") == quantity("0.002G") && quantity("2000k") == quantity("2M") && quantity("0.002G") == quantity("2000k")`,
+			expectValue: true,
+		},
+		{
+			name:        "inequality",
+			expr:        `quantity("200M") == quantity("0.3G")`,
+			expectValue: false,
+		},
+		{
+			name: "quantity_less",
+			expr: `quantity("50M").isLessThan(quantity("50Mi"))`,
+		},
+		{
+			name: "quantity_less_obvious",
+			expr: `quantity("50M").isLessThan(quantity("100M"))`,
+		},
+		{
+			name:        "quantity_less_false",
+			expr:        `quantity("100M").isLessThan(quantity("50M"))`,
+			expectValue: false,
+		},
+		{
+			name: "quantity_greater",
+			expr: `quantity("50Mi").isGreaterThan(quantity("50M"))`,
+		},
+		{
+			name: "quantity_greater_obvious",
+			expr: `quantity("50Mi").isGreaterThan(quantity("100Mi"))`,
+		},
+		{
+			name:        "quantity_greater_false",
+			expr:        `quantity("50M").isGreaterThan(quantity("100M"))`,
+			expectValue: false,
+		},
+		{
+			name:        "compare_equal",
+			expr:        `quantity("200M").compareTo(quantity("0.2G"))`,
+			expectValue: 0,
+		},
+		{
+			name:        "compare_less",
+			expr:        `quantity("50M").compareTo(quantity("50Mi"))`,
+			expectValue: -1,
+		},
+		{
+			name:        "compare_greater",
+			expr:        `quantity("50Mi").compareTo(quantity("50M"))`,
+			expectValue: 1,
+		},
+		{
+			name: "add_quantity",
+			expr: `quantity("50k").add(quantity("20")) == quantity("50.02k")`,
+		},
+		{
+			name: "add_int",
+			expr: `quantity("50k").add(20).isLessThan(quantity("50020"))`,
+		},
+		{
+			name: "sub_quantity",
+			expr: `quantity("50k").sub(quantity("20")) == quantity("49.98k")`,
+		},
+		{
+			name: "sub_int",
+			expr: `quantity("50k").sub(20) == quantity("49980")`,
+		},
+		{
+			name:        "arith_chain_1",
+			expr:        `quantity("50k").add(20).sub(quantity("100k")).asInteger()`,
+			expectValue: -49980,
+		},
+		{
+			name:        "arith_chain",
+			expr:        `quantity("50k").add(20).sub(quantity("100k")).sub(-50000).asInteger()`,
+			expectValue: 20,
+		},
+		{
+			name:        "as_integer",
+			expr:        `quantity("50k").asInteger()`,
+			expectValue: 50000,
+		},
+		{
+			name:               "as_integer_error",
+			expr:               `quantity("9999999999999999999999999999999999999G").asInteger()`,
+			expectedRuntimeErr: `cannot convert value to integer`,
+		},
+		{
+			name:        "is_integer",
+			expr:        `quantity("9999999999999999999999999999999999999G").isInteger()`,
+			expectValue: false,
+		},
+		{
+			name:        "as_float",
+			expr:        `quantity("50.703k").asApproximateFloat()`,
+			expectValue: types.Double(50703),
+		},
+	}
```

Comment:
Missing test case for isInteger return true. Also should we add expectValue to all test cases to check the eval result?

Reply:
added